### DISCUSSION TITLE
How to detect restart of a uService?

### DISCUSSION THREAD

**Discussion by ALPayne18** [eclipse-uprotocol/up-spec#168] [General]
Let's say we have the following interaction between a client uE (uClient) and service uE (uService).

1. uClient sends an RPC request to uService to start some action (example play audible tone)
2. uService begins the action and sends back an RPC response indicating success. 
3. At a later point in time, uClient sends a second RPC request for uService to stop the action (example stop playing audible tone)
4. The uService stops the action and sends back an RPC response indicating success

This is the nominal case. 

Now what happens if between steps 2 and 3, the uService goes down or is restarted? Is there any mechanism provided by uProtocol to allow the uClient to know this happened? The uClient might want to restart at 1 once the service restarted. It also might want to do an alternate action while uService is unavailable.

I glanced through the spec, and I did not see any mechanism that would allow this level of visibility. Thoughts?

**Comment by stevenhartley** [comment]
Please see #148 where we hope to discuss and address this topic.
